docs/upgrading-to-datastreams.asciidoc

```diff
@@ -0,0 +1,98 @@
+[[upgrade-to-data-streams]]
+=== Upgrade to data streams
+
+* <<why-upgrade-to-data-streams>>
+* <<considerations-data-streams>>
+* <<apm-data-streams-upgrade-steps>>
+* <<apm-data-streams-upgrade-steps-ess>>
+
+Classic indices are deprecated starting in version 7.16.0.
+This means that APM Server will no longer perform any index management -- index templates, ingest pipelines, and ILM policies will all be managed by {fleet}.
+
+Users that are not ready to <<upgrade-to-apm-integration>> should
+upgrade to data streams in the meantime.
+
+[discrete]
+[[why-upgrade-to-data-streams]]
+=== Why upgrade?
+
+Data streams offer a number of benefits over classic indices:
+
+**Data streams**:
+
+* Reduced number of fields per index, better space efficiency, and faster queries
+* More granular data control
+* Errors and metrics data streams are shared with other data sources -- which means better long-term integration with the logs and metrics apps
+* Removes template inheritance for ILM policies and makes use of new {es} index and component templates
+* Fixes <<server-resource-exists-not-alias>>
+
+[discrete]
+[[considerations-data-streams]]
+=== Data streams considerations
```

Comment:
Identified that sourcemaps are not working with data streams in `7.16` - can we add that as known-limitation please. (Fixed in 8.0).

Reply:
Added.